src/server/plugins/engine/models/FormModel.ts

```diff
@@ -289,13 +290,17 @@ export class FormModel {
     // Format relevant state errors
     const errors = error?.details.map(getError)
     const paths: string[] = []
+    const { isPreview } = checkFormStatus(request.path)
 
     // Add paths for navigation
     for (const { collection, path } of context.relevantPages) {
       paths.push(path)
 
       // Stop at current page or with errors
-      if (path === currentPath || collection.getErrors(errors)) {
+      if (
+        !isPreview &&
+        (path === currentPath || collection.getErrors(errors))
+      ) {
```

Comment:
Looks ace @davidjamesstone

Can we remove the current path check mentioned in https://github.com/DEFRA/forms-runner/pull/572#discussion_r1875573566?

Think we're covered by the loop above

```suggestion
      if (!isPreview && collection.getErrors(errors)) {
```